src/modules/quill-toolbar-accessibility-module.ts

```diff
@@ -0,0 +1,153 @@
+import { EnvironmentUtils } from "andculturecode-javascript-core";
+import { BrowserUtils } from "andculturecode-javascript-core";
+import { CoreUtils } from "andculturecode-javascript-core";
+import { Quill } from "quill";
+import { StringUtils } from "andculturecode-javascript-core";
+import { QuillModule } from "./quill-module";
+
+/**
+ * Module to hack accessibility into the toolbar.
+ * Code based on https://github.com/quilljs/quill/issues/1173#issuecomment-265912867
+ * This module can be removed once native accessibility is added to Quill core.
+ * @see https://github.com/quilljs/quill/issues/1173
+ * @see https://github.com/quilljs/quill/issues/2038
+ */
+export class QuillToolbarAccessibilityModule extends QuillModule<{}> {
+    public static modulePath = "modules/toolbarAccessibility";
+
+    constructor(quill: Quill) {
+        super(quill, {});
+        this.applyAccessibleToolbarButtonTitles();
+    }
+
+    private getToolbar(): Element | undefined {
+        const toolbars = this.quill.root.parentElement?.parentElement?.getElementsByClassName(
+            "ql-toolbar"
+        );
+
+        // if 0 or >1 toolbars, something is wrong, bail
+        if (toolbars == null || toolbars.length !== 1) {
+            return undefined;
+        }
+
+        return toolbars[0];
+    }
+
+    private async waitForToolbarToBeRendered() {
+        let retryCount = 0;
+        let toolbar = this.getToolbar();
+        while (toolbar == null && retryCount < 10) {
+            await CoreUtils.sleep(100);
+            toolbar = this.getToolbar();
+            retryCount++;
+        }
+
+        if (toolbar == null) {
+            EnvironmentUtils.runIfDevelopment(() =>
+                console.warn("Quill toolbar not rendered after 1 second.")
+            );
+        }
+    }
+
+    private async applyAccessibleToolbarButtonTitles() {

```

Comment:
There's a lot going on in this function - is there any way it can be broken down into multiple functions?